particle-iot/device-os
# [deprecation] remove setup_done flag and add deprecation notice

### Description

This PR removes the setup-done flag, and adds deprecation notice when the dct setup-done flag is used.

#### Starting Device OS 4.x:
1. Attempting to set this bit via particle CLI or BLE will not have any impact on the device behavior. The setup_done bit can be modified only via user app.
2. Setting this via user app causes the following deprecation warning. Starting Device OS 5.x, the setup_done bit and the relevant HAL flag will be completely removed and will cause a build error.
```
Setup Done flag has been deprecated since Device OS 4.0 and has no effect on device behavior. This will be removed in Device OS 5.x causing build error!
```

### Steps to Test

Run an app that reads/writes `DCT_SETUP_DONE_OFFSET` and verify that the following warning occurs during compiling
`Setup Done flag has been deprecated since Device OS 4.0 and has no effect on device behavior`

### Example App
```
#include "Particle.h"
#include "dct.h"
SYSTEM_MODE(SEMI_AUTOMATIC);
void setup() {
	// 0x01 = setup done
	// 0xff = setup not done (go into listening mode at boot)
	const uint8_t val = 0x01;
	dct_write_app_data(&val, DCT_SETUP_DONE_OFFSET, 1);

	// This is just so you know the operation is complete
	pinMode(D7, OUTPUT);
	digitalWrite(D7, HIGH);
}

void loop() {
}
```

2. The following commands shouldn't break when run through CLI
![Screen Shot 2022-04-29 at 12 15 38 PM](https://user-images.githubusercontent.com/48732259/166009026-e737a210-0dc7-40fc-b607-910ffec94910.png)



3. Mobile app should setup OK on factory fresh module

### References

[SC-72116](https://app.shortcut.com/particle/story/72116/remove-the-need-for-setup-done-setupdone-flag-in-device-os-across-the-board-for-all-devices-sims)

---

### Completeness
